src/NUnitFramework/framework/Internal/Execution/WorkItemQueue.cs

```diff
@@ -244,8 +271,11 @@ public WorkItem Dequeue()
 
 
                 // Dequeue our work item
-                WorkItem work;
-                while (!_innerQueue.TryDequeue(out work)) { };
+                WorkItem work = null;
+                while (work == null)
+                    foreach (var q in _innerQueues)
+                        if (q.TryDequeue(out work))
+                            break;
```

Comment:
I don't believe so. The old code also had a loop using TryDequeue(), so it kept looping until something was found in the queue...

```C#
while (!_innerQueue.TryDequeue(out work){ }
```

In this case, we want to try two queues. I could have written...

```C#
while (!_innerQueues[0].TryDequeue(out work) && !_innerQueues[1](out work){ }
```

However, I wanted to keep the number of priorities open, at least in theory, so I used the loop.

What the SpinWait does is ensure that only one thread can be in this code at a single time. Another thread will need to come along and add a `WorkItem` to one of the queues, of course.